lighthouse-core/gather/gather-runner.js

```diff
@@ -233,6 +258,7 @@ class GatherRunner {
 
     const networkError = GatherRunner.getNetworkError(mainRecord);
     const interstitialError = GatherRunner.getInterstitialError(mainRecord, networkRecords);
+    const docTypeError = GatherRunner.getDocTypeError(mainRecord);
```

Comment:
The structure of this function is rather strange–zooming out a bit, I'd expect the `loadFailureMode` check to happen first; and for each of these errors to be created in order, but only if the former error check didn't hit. Incrementally, what you have here makes sense, I just think this entire function might benefit from a slight refactor.

Reply:
Yeah, there is also definitely some redundant checks in each get function that don't need to be there if it was refactored. Should I try to refactor in this PR? It looks like Patrick wrote this function originally, I can also ping them about this if required.

Reply:
@connorjclark is the refactoring we're talking about here moving 1 line a few lines up and 1 more line a few lines down? I think we can manage it here 😉 

be my guest! I certainly have no strong attachment to the specific order in which they're declared as long as we don't let them become multi-line statements :)

Reply:
> moving 1 line a few lines up and 1 more line a few lines down?

yeah, that sounds like no problem, but I would be hesitant to do much deduplicating of content within the error checks. They work well today and it's easy to reason about them since they're more or less independent of each other, each just checking on their own particular error case.

Maybe pulling out the `!mainRecord` check makes sense since they all end up having to check that and the type system will keep track of it already being done (since we can drop all the `|undefined`s)? But, again, everything's working today and this function isn't in bad shape or anything, so @lemcardenas you can also feel free to leave it to a follow-up PR or to someone else in the future if it feels like the PR scope is growing too much.

https://google.github.io/eng-practices/review/developer/small-cls.html is one of my favorite docs :)

Reply:
I think I'll try to stay safe for this PR and won't change anything about the getPageLoadError function, because as mentioned below there may be more work needed for the original change to work.